### PR TITLE
RevEng: Use unbound NavProps on the IModel instead of annotations

### PR DESCRIPTION
**Pull request by lajones**
Fix for #5154. This is clean up - no change to output from Scaffolding. Now that you can add "shadow" NavProps to an `IModel` we make use of them so we can stop using the artificial `ScaffoldingForeignKeyAnnotations`.

Change to `ForeignKey` is just cleaning up a whitespace problem I saw when I was doing this.

### CODE REVIEW

**Comment by lajones** [comment]
Note for provider-writers: if you previously were using `ScaffoldingForeignKeyAnnotations` to store navigation property names, you'll have to update to use `foreignKey.HasDependentToPrincipal()` and `foreignKey.HasPrincipalToDependent()` instead.

File: src/Microsoft.EntityFrameworkCore.Tools.Core/Scaffolding/Configuration/Internal/ModelConfiguration.cs [hunk 1]
```diff
@@ -674,10 +673,8 @@ public virtual void AddRelationshipConfiguration([NotNull] EntityConfiguration e
 
             foreach (var foreignKey in entityConfiguration.EntityType.GetForeignKeys())
             {
-                var dependentEndNavigationPropertyName =
-                    foreignKey.Scaffolding().DependentEndNavigation;
-                var principalEndNavigationPropertyName =
-                    foreignKey.Scaffolding().PrincipalEndNavigation;
+                var dependentEndNavigationPropertyName = foreignKey.DependentToPrincipal.Name;
+                var principalEndNavigationPropertyName = foreignKey.PrincipalToDependent.Name;
```
Comment (smitpatel) [inline src/Microsoft.EntityFrameworkCore.Tools.Core/Scaffolding/Configuration/Internal/ModelConfiguration.cs]: null check for navigations needed?

**Comment by smitpatel** [comment]
LGTM.

File: src/Microsoft.EntityFrameworkCore.Tools.Core/Scaffolding/Configuration/Internal/ModelConfiguration.cs (hunk 1 above)
Comment (lajones) [inline src/Microsoft.EntityFrameworkCore.Tools.Core/Scaffolding/Configuration/Internal/ModelConfiguration.cs]: Will do. After that can I convert the LGTM to a ship-it?

**Comment by smitpatel** [comment]
If someone else wants to review otherwise :shipit:

**Comment by lajones** [comment]
Checked in with commit 767b3dd.

**Pull request closed without merging**